src/plugins/data_view_field_editor/public/components/field_editor/field_editor.tsx

```diff
@@ -189,6 +152,89 @@ const FieldEditorComponent = ({ field, onChange, onFormModifiedChange }: Props)
 
   const isValueVisible = get(formData, '__meta__.isValueVisible');
 
+  const lastPreview$ = useMemo(() => {
+    const replaySubj = new BehaviorSubject<FieldPreview[]>([]);
+    fieldPreview$.subscribe(replaySubj);
+    return replaySubj;
+  }, [fieldPreview$]);
+
+  const resetTypes = useMemo(
+    () => () => {
+      const lastVal = lastPreview$.getValue();
+      // resets the preview history to an empty set
+      fieldPreview$.next([]);
+      // apply the last preview to get all the types
+      fieldPreview$.next(lastVal);
+    },
+    [fieldPreview$, lastPreview$]
+  );
+
+  const lastPreview = useRef<FieldPreview[]>();
+
+  useEffect(() => {
+    const existingCompositeField = !!Object.keys(form.getFormData().fields || {}).length;
+
+    const subLastPreview = fieldPreview$.subscribe((val) => {
+      lastPreview.current = val;
+    });
+    const changes$ = fieldPreview$.pipe(
+      map((items) =>
```

Comment:
It would be great to move this transformation chain outside the react component and test separately. Seems like it doesn't depend on anything 

it could look like this in the end: 

```
const changes$ = getFieldPreviewChanges(fieldPreview$)
// or
const changes$ = fieldPreview.pipe(getFieldPreviewChanges())
```

Reply:
working on this